sdk/storage/azure-storage-file-datalake/tests/test_file_system.py

```diff
@@ -87,6 +84,153 @@ def test_create_file_system_extra_backslash(self, datalake_storage_account_name,
+
+    @pytest.mark.live_test_only
+    @DataLakePreparer()
+    def test_create_file_system_encryption_scope_file_system_sas(self, datalake_storage_account_name, datalake_storage_account_key):
+        self._setUp(datalake_storage_account_name, datalake_storage_account_key)
+        # Arrange
+        url = self.account_url(datalake_storage_account_name, 'dfs')
+        file_system_name = self._get_file_system_reference()
+        token = generate_file_system_sas(
+            self.dsc.account_name,
+            file_system_name,
+            self.dsc.credential.account_key,
+            permission=FileSystemSasPermissions(write=True, read=True, delete=True),
+            expiry=datetime.utcnow() + timedelta(hours=5),
+            encryption_scope="hnstestscope1")
+        encryption_scope = FileSystemEncryptionScope(default_encryption_scope="hnstestscope1")
+
+        # Act
+        file_system_client = self.dsc.get_file_system_client(file_system_name)
+        file_system_client.create_file_system(file_system_encryption_scope=encryption_scope)
+
+        fsc_sas = FileSystemClient(url, file_system_name, token)
+        fsc_sas.create_file('file1')
+        fsc_sas.create_directory('dir1')
+        dir_props = fsc_sas.get_directory_client('dir1').get_directory_properties()
+        file_props = fsc_sas.get_file_client('file1').get_file_properties()
+
+        # Assert
+        self.assertTrue(dir_props)
+        self.assertIsNotNone(dir_props.encryption_scope)
+        self.assertEqual(dir_props.encryption_scope, encryption_scope.default_encryption_scope)
+        self.assertTrue(file_props)
+        self.assertIsNotNone(file_props.encryption_scope)
+        self.assertEqual(file_props.encryption_scope, encryption_scope.default_encryption_scope)
+
+    @pytest.mark.live_test_only
+    @DataLakePreparer()
+    def test_create_file_system_encryption_scope_directory_sas(self, datalake_storage_account_name, datalake_storage_account_key):
+        self._setUp(datalake_storage_account_name, datalake_storage_account_key)
+        # Arrange
+        url = self.account_url(datalake_storage_account_name, 'dfs')
+        file_system_name = self._get_file_system_reference()
+        token = generate_directory_sas(
+            self.dsc.account_name,
+            file_system_name,
+            'dir1',
+            self.dsc.credential.account_key,
+            permission=FileSasPermissions(write=True, read=True, delete=True),
+            expiry=datetime.utcnow() + timedelta(hours=5),
+            encryption_scope="hnstestscope1")
+        encryption_scope = FileSystemEncryptionScope(default_encryption_scope="hnstestscope1")
+
+        # Act
+        file_system_client = self.dsc.get_file_system_client(file_system_name)
+        file_system_client.create_file_system(file_system_encryption_scope=encryption_scope)
+
+        fsc_sas = FileSystemClient(url, file_system_name, token)
```

Comment:
Hmm typically you wouldn't create a file system with a directory/file SAS since technically it wouldn't work for file system operations but I guess in this case, since you are only doing directory things with this client, it works.

I think maybe it would be better to create a `DirectoryClient` here? and a `FileClient` below? Just so the SAS usage makes a bit more sense.

Reply:
Ohh gotcha. Yes I can make this adjustment!